tests/acceptance/actions-test.js

```diff
@@ -48,6 +54,20 @@ test('allows to chain actions', function(assert) {
   });
 });
 
+test('allows to chain aliased actions', function(assert) {
```

Comment:
@magistrula now I see what the problem is:

When you invoke an action, the action returns the current page object context so you can chain other actions. The problem here arises when nesting properties:

```js
const page = create({
  foo: {
  },

  bar: clickable('btn')
});

page.foo.clickOn('lorem ipsum').bar(); // fails
```

The call to `clickOn` returns the object `foo` and not the root page object, so `bar` is not found. This is the default behavior of the page object and it has been working fine so far.

The question is, how should `alias` behave when you use a nested action? should it return the aliased action context or the alias context?

```js
const page = create({
  foo: {
  },

  bar: clickable('btn'),

  fooClickAlias: alias('foo.clickOn')
});

page.fooClickAlias('lorem ipsum').bar(); // fails
```

My guess would be that it should return whatever the action returns (because is just a function after all).

So, updating the acceptance test to something like this should be enough

```diff
--- a/tests/acceptance/actions-test.js
+++ b/tests/acceptance/actions-test.js
@@ -18,16 +18,16 @@ let page = PageObject.create({
   keys: {
     clickOn: clickOnText('.numbers'),
     sum: clickable('button', { scope: '.operators', at: 0 }),
-    equal: clickable('button', { scope: '.operators', at: 2 })
+    equal: clickable('button', { scope: '.operators', at: 2 }),
+    add: alias('sum'),
+    getResult: alias('equal')
   },

   screen: value('.screen input'),
   fillValue: fillable('.screen input'),

   visitCalculator: alias('visit'),
-  clickOnKey: alias('keys.clickOn'),
-  add: alias('keys.sum'),
-  getResult: alias('keys.equal')
-  getResult: alias('keys.equal')
+  clickOnKey: alias('keys.clickOn')
 });

 test('allows to chain actions', function(assert) {
 });

 test('allows to chain actions', function(assert) {
@@ -58,13 +58,12 @@ test('allows to chain aliased actions', function(assert) {
   page
     .visitCalculator()
     .clickOnKey('1')
-    .clickOnKey('2')
-    .add()
-    .clickOnKey('3')
+    .sum()
+    .clickOn('2')
     .getResult();

   andThen(function() {
-    assert.equal(page.screen, '15');
+    assert.equal(page.screen, '3');
   });
 });
```

---

To continue with the other discussion, if `alias` should change or not the return value of the actions, we could add an option to `alias` in the future if we see the need for it, something like

```js
const page = create({
  foo: {
  },

  bar: clickable('btn'),

  fooClickAlias: alias('foo.clickOn', { returnSelf: true })
});

page.fooClickAlias('lorem ipsum').bar(); // works
```

What do you think?